src/autolabel/tasks/classification.py

```diff
@@ -55,6 +60,40 @@ def construct_prompt(self, input: Dict, examples: List) -> str:
         # prepare task guideline
         labels_list = self.config.labels_list()
         num_labels = len(labels_list)
+
+        # if large number of labels, filter labels_list by similarity of labels to input
+        if num_labels >= 50:
+            example_prompt = PromptTemplate(
+                input_variables=["input"],
+                template="{input}",
+            )
+            label_examples = [{"input": label} for label in labels_list]
+
+            example_selector = SemanticSimilarityExampleSelector.from_examples(
```

Comment:
instead of creating this each time, is it possible to construct this example selector once and then just call sample labels each time. This would make sure that we just embed the label list once.

Reply:
Good idea.

I now construct the LabelSelector once in agent.run() and agent.plan(). This way embeddings of labels are only computed once.